### PR TITLE
release-21.1: Backport changefeed observability PRs.

### PR DESCRIPTION
**Pull request by miretskiy**
Backport:
  * 4/4 commits from "changefeed,server: changefeed histograms and configurable slow logging" (#63923)
  * 2/2 commits from " changefeedccl: changefeed concurrency and frontier observability #67206" (#67268)

Please see individual PRs for details.

/cc @cockroachdb/release

Release Justification: A low danger, high impact observability change.  The added metrics allow us to troubleshoot
changefeed performance, when running in large scale clusters.

### CODE REVIEW

File: pkg/settings/setting.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 
 // MaxSettings is the maximum number of settings that the system supports.
 // Exported for tests.
-const MaxSettings = 512
+const MaxSettings = 288
```
Comment (stevendanna) [inline pkg/settings/setting.go:23]: Guessing we don't want to decrease this :D

**Review by stevendanna — CHANGES_REQUESTED** [review]

File: pkg/settings/setting.go (hunk 1 above)
Comment (miretskiy) [reply pkg/settings/setting.go:23]: is this persisted somewhere?

Comment (stevendanna) [reply pkg/settings/setting.go]: Not persisted, but if we register more settings than the max a panic is thrown.  My guess is a few PRs bumped it when we hit the old limit, so we should probably stick with the higher value.

Comment (miretskiy) [reply pkg/settings/setting.go]: Fixed. Thanks.

Comment (miretskiy) [reply pkg/settings/setting.go]: Removed one of the commits since the setting was increased multiple times.

**Review by stevendanna — APPROVED** [review]
The metrics changes have been baking a while and are pretty low risk.  The changes to the export request limit seems pretty straightforward and high value for large clusters.

**Pull request merged**